src/bond.ts

```diff
@@ -568,57 +623,264 @@ export class BondContract extends Contract {
     return setContractStateTx;
   }
 
+  /**
+   *Builds a transaction to blacklist addresses.
+   * @param senderAddress - The address of the sender.
+   * @param addresses - An array of addresses to be removed from the blacklist.
+   */
   setAcceptedCallers(senderAddress: IAddress, addresses: IAddress[]) {
-    throw new Error('Not implemented');
+    const inputAddresses = addresses.map(
+      (address) => new AddressValue(address)
+    );
+    const tx = new Transaction({
+      value: 0,
+      data: new ContractCallPayloadBuilder()
+        .setFunction('setBlacklist')
+        .setArgs(inputAddresses)
+        .build(),
+      receiver: this.contract.getAddress(),
+      sender: senderAddress,
+      gasLimit: 20_000_000,
+      chainID: this.chainID
+    });
+    return tx;
   }
 
+  /**
+   *Builds a transaction to blacklist addresses.
+   * @param senderAddress - The address of the sender.
+   * @param addresses - An array of addresses to be removed from the blacklist.
+   */
   setBlacklist(senderAddress: IAddress, addresses: IAddress[]) {
-    throw new Error('Not implemented');
+    const inputAddresses = addresses.map(
+      (address) => new AddressValue(address)
+    );
+    const tx = new Transaction({
+      value: 0,
+      data: new ContractCallPayloadBuilder()
+        .setFunction('setBlacklist')
+        .setArgs(inputAddresses)
+        .build(),
+      receiver: this.contract.getAddress(),
+      sender: senderAddress,
+      gasLimit: 20_000_000,
+      chainID: this.chainID
+    });
+    return tx;
   }
 
-  removeBlacklist(senderAddress: IAddress, addresses: IAddress[]) {
-    throw new Error('Not implemented');
+  /**
+   *Builds a transaction to remove addresses from the blacklist.
+   * @param senderAddress - The address of the sender.
+   * @param compensationId compensaton id to check if exists
+   * @param addresses - An array of addresses to be removed from the blacklist.
+   */
+  removeBlacklist(
+    senderAddress: IAddress,
+    compensationId: number,
+    addresses: IAddress[]
+  ) {
+    const toBeRemovedAddresses = addresses.map(
+      (address) => new AddressValue(address)
+    );
+    const tx = new Transaction({
+      value: 0,
+      data: new ContractCallPayloadBuilder()
+        .setFunction('removeBlacklist')
+        .addArg(new U64Value(compensationId))
+        .setArgs(toBeRemovedAddresses)
+        .build(),
+      receiver: this.contract.getAddress(),
+      sender: senderAddress,
+      gasLimit: 20_000_000,
+      chainID: this.chainID
+    });
+    return tx;
   }
 
+  /**
+   * Builds a transaction to remove the accepted callers from the sender's address.
```

Comment:
```suggestion
   * Builds a transaction to remove the accepted callers 
```

Reply:
Worth keeping some consistency like in the other methods.

```suggestion
   * Builds a `removeAcceptedCallers` transaction (... further description for the endpoint, if needed)
```